lib/block/header.ex

```diff
@@ -0,0 +1,370 @@
+defmodule Block.Header do
+  @moduledoc """
+  This structure codifies the header of a block in the blockchain.
+  """
+
+  # The start of the Homestead block, as defined in Eq.(13) of the Yellow Paper (N_H)
+  @homestead 1_150_000
+  @empty_trie MerklePatriciaTree.Trie.empty_trie()
+
+  defstruct [
+    parent_hash: <<>>,               # Hp P(BH)Hr
+    ommers_hash: @empty_trie,        # Ho KEC(RLP(L∗H(BU)))
+    beneficiary: <<>>,               # Hc
+    state_root: @empty_trie,         # Hr TRIE(LS(Π(σ, B)))
+    transactions_root: @empty_trie,  # Ht TRIE({∀i < kBTk, i ∈ P : p(i, LT (BT[i]))})
+    receipts_root: @empty_trie,      # He TRIE({∀i < kBRk, i ∈ P : p(i, LR(BR[i]))})
+    logs_bloom: <<>>,                # Hb bloom
+    difficulty: nil,                 # Hd
+    number: nil,                     # Hi
+    gas_limit: 0,                    # Hl
+    gas_used: 0,                     # Hg
+    timestamp: nil,                  # Hs
+    extra_data: <<>>,                # Hx
+    mix_hash: nil,                   # Hm
+    nonce: nil,                      # Hn
+  ]
+
+  # As defined in Eq.(35)
+  @type t :: %__MODULE__{
+    parent_hash: EVM.hash | <<>>,
+    ommers_hash: EVM.trie_root,
+    beneficiary: EVM.address | <<>>,
+    state_root: EVM.trie_root,
+    transactions_root: EVM.trie_root,
+    receipts_root: EVM.trie_root,
+    logs_bloom: binary(), # TODO
+    difficulty: integer() | nil,
+    number: integer() | nil,
+    gas_limit: EVM.val,
+    gas_used: EVM.val,
+    timestamp: EVM.timestamp | nil,
+    extra_data: binary(),
+    mix_hash: EVM.hash | nil,
+    nonce: <<_::64>> | nil, # TODO: 64-bit hash?
+  }
+
+  @d_0 131_072 # Eq.(40)
```

Comment:
Can we call this @homestead_block_number or something similar with a comment that it's `d0`?